### PR TITLE
RD-3241 Added edit mode support for page groups (excluding renames)

### PR DESCRIPTION
**Pull request by kubama**
## Description
Added edit mode support for page groups.

**TODO:**
- Add support for renaming menu items (inline edit with a triggering icon) - I'm planning to reuse `EditableLabel` component but it needs to be extended to be able to work with a triggering icon - now it only supports label click. Change in `cloudify-ui-components` will be needed.
- Extend component test - I'm planning to try out drag'n'drop plugin for cypress

## Screenshots / Videos
CPT2110081801-183x431

## Checklist

- [X] My code follows the UI Code Style.
- [X] I verified that all tests and checks have passed.
- [X] I verified if that change requires a documentation update.
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/1366/
Re-run of the failed login spec: https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/1369

I consider it a pass ✔️.

## Documentation
Will need documentation update once feature implementation is complete.

### CODE REVIEW

**Comment by kubama** [comment]
> 👍🏻 By feature you mean page groups or all changes related to new menus?

I mean all planned changes. I assume the updates will include screenshots, so it makes sense to wait until the look is finalized.

> Do you maybe have a list of places to update already?

No, I didn't look for it yet.

**Comment by qooban** [comment]
> > 👍🏻 By feature you mean page groups or all changes related to new menus?
> 
> I mean all planned changes. I assume the updates will include screenshots, so it makes sense to wait until the look is finalized.
> 
> > Do you maybe have a list of places to update already?
> 
> No, I didn't look for it yet.

OK, so I have created RD-3341 for documentation update and added there some points (the list as mentioned can be updated during epic development, it is not finalized).

File: app/components/sidebar/AddPageGroupButton.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import i18n from 'i18next';
+
+import { useDispatch } from 'react-redux';
+import EditModeButton from '../EditModeButton';
+import { addPageGroup } from '../../actions/page';
+
+let nameIndex = 0;
+
+const StyledEditModeButton = styled(EditModeButton)`
+    &&&&&& {
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/components/sidebar/AddPageGroupButton.tsx:14]: Added

File: app/actions/types.ts [hunk 2]
```diff
@@ -39,9 +39,9 @@ export const REMOVE_WIDGET = 'REMOVE_WIDGET';
 export const MINIMIZE_WIDGETS = 'MINIMIZE_WIDGETS';
 export const MINIMIZE_TAB_WIDGETS = 'MINIMIZE_TAB_WIDGETS';
 export const UPDATE_WIDGET = 'UPDATE_WIDGET';
-export const REMOVE_PAGE = 'REMOVE_PAGE';
+export const REMOVE_PAGE_MENU_ITEM = 'REMOVE_PAGE';
 export const CHANGE_PAGE_DESCRIPTION = 'CHANGE_PAGE_DESCRIPTION';
-export const REORDER_PAGE = 'REORDER_PAGE';
+export const REORDER_PAGE_MENU = 'REORDER_PAGE';
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/actions/types.ts:44]: Updated

File: app/components/sidebar/PagesList.tsx [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+import {
+    createPagesMap,
+    InsertPosition,
+    removePageWithChildren,
+    removeSinglePageMenuItem,
+    reorderPageMenu,
+    selectPage
+} from '../../actions/page';
+import AddPageGroupButton from './AddPageGroupButton';
+import { useBoolean } from '../../utils/hooks';
```
[the remark this answers is not in the document]
Comment (qooban) [reply app/components/sidebar/PagesList.tsx:26]: So maybe let's reorder them as this file is already recognized as new.
We can maybe have components imports, utils, actions?

File: app/actions/types.ts [hunk 4]
```diff
@@ -39,9 +39,9 @@ export const REMOVE_WIDGET = 'REMOVE_WIDGET';
 export const MINIMIZE_WIDGETS = 'MINIMIZE_WIDGETS';
 export const MINIMIZE_TAB_WIDGETS = 'MINIMIZE_TAB_WIDGETS';
 export const UPDATE_WIDGET = 'UPDATE_WIDGET';
-export const REMOVE_PAGE = 'REMOVE_PAGE';
+export const REMOVE_PAGE_MENU_ITEM = 'REMOVE_PAGE';
```
Comment (qooban) [inline app/actions/types.ts:42]: Did you consider changing value of this const as well as the other (`REORDER_PAGE_MENU`)?

File: app/components/sidebar/PagesList.tsx [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    pages: PageMenuItem[];
+    selected?: string;
+    isEditMode: boolean;
+}
+
+function drillDownPagesFilter(pageMenuItem: PageMenuItem) {
+    return pageMenuItem.type === 'pageGroup' || !pageMenuItem.isDrillDown;
+}
+
+const PagesList: FunctionComponent<PagesListProps> = ({
```
[the remark this answers is not in the document]
Comment (qooban) [reply app/components/sidebar/PagesList.tsx:40]: OK 👍🏻

**Review by qooban — DISMISSED** [review]
Looks good overall 👍🏻 

Maybe some small changes can be implemented (as in comments).

File: app/actions/types.ts (hunk 4 above)
Comment (kubama) [reply app/actions/types.ts]: Updated

File: app/components/sidebar/AddPageButton.tsx [hunk 6]
```diff
@@ -29,6 +29,7 @@ const AddPageButton = ({ onClick }) => {
             onClick={onClick}
             content={i18n.t('editMode.addPage', 'Add Page')}
             className="addPageBtn"
+            style={{ width: 146 }}
```
[the remark this answers is not in the document]
Comment (qooban) [reply app/components/sidebar/AddPageButton.tsx]: Got it. I was just not sure about responsiveness of such design. 
For now I think we can leave it as there will be look'n'feel change anyway in the future.

**Pull request merged**